File: cs50/pset6/sentimental/mario/mario.py
def validate_input(x):
    if(x < 0 or x > 23 or not isinstance(x, int)):
        return False
    else:
        return True


def pyramid(x):
    count = 0
    while count < x:
        print(" " * (x - 1 - count) + "#" * (count + 1) + "  " + "#" * (count + 1))
        count += 1


def main():
    exit = False
    while not exit:
        try:
            x = input("Height: ")
            x = int(x)
        except:
            continue
        if(validate_input(x)):
            pyramid(x)
            exit = True


main()

File: cs50/pset1/mario/more/mario.c
#include <stdio.h>
#include <cs50.h>

int validate_input(int input)
{
    if(input<0 || input>23){
        return false;
    }
    else{
        return true;
    }
}

int pyramid(int input)
{
    int height = input-1;

    for(int i = 0; i<height+1; i++){
        printf("%.*s", height-(i), "                        ");
        printf("%.*s", i+1, "#############################");
        printf("  %.*s\n", i+1, "#############################");
    }
    return 0;
}

int main(void)
{
    bool exit = false;

    while(exit == false){
        int input = get_int("Height: ");
        if(validate_input(input)){
            printf("%i is a valid input\n", input);
            pyramid(input);
            exit = true;
        }
    }
}